MapboxCoreNavigation/PortableRouteController.swift

```diff
@@ -67,6 +67,24 @@ open class PortableRouteController: RouteController {
         return !offRoute
     }
     
+    /**
+     Advances the leg index. This override also advances the leg index of the native navigator.
+     
+     This is a convienence method provided to advance the leg index of any given router without having to worry about the internal data structure of the router.
+     */
+    override public func advanceLegIndex(location: CLLocation) {
+        precondition(!routeProgress.isFinalLeg, "Can not increment leg index beyond final leg.")
```

Comment:
The superclass implementation already enforces this precondition. At least this redundant check should follow the `super` call.